src/ui/views/EditAwards.football.tsx

```diff
@@ -0,0 +1,252 @@
+		if (state.aws == undefined) {
+			setState(prevState => {
+				const aws = helpers.deepCopy(awards);
+				return {
+					...prevState,
+					aws,
+				};
+			});
+		}
+		setState(prevState => {
+			const aws: any = prevState.aws;
+			if (p.currentStats == undefined) {
+				p.currentStats = { keyStats: "No stats available" };
+			}
+			if (
+				type == "finalsMvp" ||
+				type == "mvp" ||
+				type == "droy" ||
+				type == "oroy" ||
+				type == "dpoy"
+			) {
+				aws[type] = {
+					pid: p.pid,
+					name: p.name,
+					tid: p.tid,
+					abbrev: p.abbrev,
+					keyStats: p.currentStats.keyStats,
+				};
+			} else if (type == "allDefensive" || type == "allLeague") {
+				let arrayPids: number[] = [];
+				aws[type].map((team: any, index: number) => {
+					team["players"].map((element: Player, index: number) => {
+						arrayPids.push(element.pid);
+					});
+				});
+				if (arrayPids.includes(p.pid)) {
+					logEvent({
+						type: "error",
+						text: "Player already in teams of this category",
+						saveToDb: false,
+						persistent: true,
+					});
+					error = true;
+					obj.setValue(obj.state.select.value);
+					return {
+						...prevState,
+						aws,
+					};
+				}
+
+				aws[type][numberTeam]["players"][numberPlayer] = {
+					pid: p.pid,
+					name: p.name,
+					tid: p.tid,
+					abbrev: p.abbrev,
+					keyStats: p.currentStats.keyStats,
+				};
+			} else if (type == "allRookie") {
+				let arrayPids: number[] = [];
+				aws[type].map((element: any) => {
+					arrayPids.push(element.pid);
+				});
+				if (arrayPids.includes(p.pid)) {
+					logEvent({
+						type: "error",
+						text: "Player already in teams of this category",
+						saveToDb: false,
+						persistent: true,
+					});
+					error = true;
+					obj.setValue(obj.state.select.value);
+					return {
+						...prevState,
+						aws,
+					};
+				}
+				aws[type][numberPlayer] = {
+					pid: p.pid,
+					name: p.firstName + " ",
+					tid: p.tid,
+					abbrev: p.abbrev,
+					keyStats: p.currentStats.keyStats,
+				};
+			}
+
+			return {
+				...prevState,
+				aws,
+			};
+		});
+		return error;
+	};
+
+	const handleFormSubmit = async (event: FormEvent) => {
+		event.preventDefault();
+		setState(prevState => ({
+			...prevState,
+			saving: true,
+		}));
+
+		try {
+			await toWorker("main", "upsertAwards", state, awardsInitial);
+			realtimeUpdate([], helpers.leagueUrl(["history"]));
+		} catch (error) {
+			logEvent({
+				type: "error",
+				text: error.message,
+				saveToDb: false,
+				persistent: true,
+			});
+			setState(prevState => ({
+				...prevState,
+				saving: false,
+			}));
+		}
+	};
+	if (awards) {
+		return (
+			<form onSubmit={handleFormSubmit}>
+				<div className="row">
+					<div className="col-sm-3 col-6 form-group">
+						<label>Finals MVP</label>
+						<SelectReact
+							options={players}
+							player={awards["finalsMvp"]}
+							award="finalsMvp"
+							changing={handleChange.bind(this)}
```

Comment:
Don't need `.bind(this)` in any of these, because `handleChange` does not use `this` so it doesn't matter what it is. Also `this` is undefined here anyway, since this is in an arrow function. Another nice thing about not using classes, you can ignore all the nonsense about what `this` is!